samza-core/src/main/scala/org/apache/samza/job/local/ThreadJobFactory.scala

```diff
@@ -38,6 +41,14 @@ class ThreadJobFactory extends StreamJobFactory with Logging {
     info("Creating a ThreadJob, which is only meant for debugging.")
     val coordinator = JobModelManager(config)
     val jobModel = coordinator.jobModel
+    val metricsRegistry = new MetricsRegistryMap()
+
+    //create necessary checkpoint and changelog streams, if not created
+    CheckpointManagerUtil.createAndInit(coordinator.jobModel, metricsRegistry)
+    val changelogPartitionManager = ChangelogPartitionManager.fromConfig(config, getClass.getSimpleName, metricsRegistry)
```

Comment:
Is this due to the reason that you want to skip the initialization of coordinator systems? But in fromConfig() we do it anyways?

Reply:
Initially I wanted to hide the initialization, but it's probably a bad way to do it. Will fix it in refactoring the changelogManager interface.